libraries/tock-register-interface/src/fields.rs

```diff
@@ -477,6 +490,57 @@ macro_rules! register_bitmasks {
             }
         }
     };
+
+    // Implement the `RegisterDebugInfo` trait for the register. Refer to its
+    // documentation for more information on the individual types and fields.
+    (
+        // final implementation of the macro
+        @debug $valtype:ident, $reg_mod:ident, $reg_desc:ident, [$($field:ident),*]
+    ) => {
+        impl $crate::debug::RegisterDebugInfo<$valtype> for $reg_desc {
+            // Sequence of field value enum types (implementing `TryFromValue`,
+            // produced above), generated by recursing over the fields:
+            type FieldValueEnumTypes = $crate::register_bitmasks!(
+                @fv_enum_type_seq $valtype, $($field::Value),*
+            );
+
+            fn name() -> &'static str {
+                stringify!($reg_mod)
+            }
+
+            fn field_names() -> &'static [&'static str] {
+                &[
+                    $(
+                        stringify!($field)
+                    ),*
+                ]
+            }
+
+            fn fields() -> &'static [Field<$valtype, Self>] {
+                &[
+                    $(
+                        $field
+                    ),*
+                ]
+            }
+        }
+    };
+
+    // Build the recursive `FieldValueEnumSeq` type sequence:
```

Comment:
@lschuermann Do you think it would be good to include a small example of how the resulting recursion chain will look like.
Something like
```rust
FieldValueEnumCons<u32,Foo, 
    FieldValueEnumCons<u32, Bar,
        FieldValueEnumCons<u32, Baz,
            FieldValueEnumNil
        >
    >
>
```
Just to make it clear

Reply:
That's a good idea, added. These macros are hard enough to read already. :sweat_smile: